pkg/detectors/detectors.go

```diff
@@ -27,6 +28,23 @@ type Detector interface {
 	Type() detectorspb.DetectorType
 }
 
+// ConditionalDetector is an optional interface that a detector can implement to
+// skip chunks based on specific criteria.
+type ConditionalDetector interface {
+	// ScanChunk determines whether the detector should run.
+	ScanChunk(chunk sources.Chunk) bool
```

Comment:
this is minor, but i double-took at the verb `Scan` here. i expected something like `ShouldScanChunk` instead.

Reply:
It was originally `ShouldScanChunk`, I'm not sure why I removed `Should`; naming things is hard.

Do you prefer `ShouldScanChunk` or `MatchesChunk`?

Reply:
My concern with `MatchesChunk` is that the term "matches" is used elsewhere in detector code and novices might be tempted to conflate the two senses. I think `ShouldScanChunk` is fine and the least likely to confuse.

Reply:
Makes sense. I've changed the name.